### PR TITLE
Add switch ASIC vendor and platform for Nephos

### PR DESCRIPTION
**Pull request by kaiyu22**
**- What I did**
    Add switch ASIC vendor: Nephos
    Add Nephos platform: Ingrasys S9130-32X

**- How I did it**
    Add platform/nephos files
    Add platform/nephos/sonic-platform-modules-ingrasys submodule
    Add device/ingrasys/x86_64-ingrasys_s9130_32x-r0 files
    Add SONiC to support Nephos platform

**- How to verify it**
    To build SONiC installer image and docker images, run the following commands:
       make configure PLATFORM=nephos
       make target/sonic-nephos.bin
    Check system and network feature is worked as well

**- Description for the changelog**
    Add switch ASIC vendor and platform for Nephos

**- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: Sam Yang <yang.kaiyu@gmail.com>

### CODE REVIEW

File: platform/nephos/rules.mk [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+include $(PLATFORM_PATH)/docker-syncd-nephos.mk
+include $(PLATFORM_PATH)/docker-syncd-nephos-rpc.mk
+include $(PLATFORM_PATH)/one-image.mk
+include $(PLATFORM_PATH)/one-aboot.mk
+include $(PLATFORM_PATH)/libsaithrift-dev.mk
+include $(PLATFORM_PATH)/python-saithrift.mk
+include $(PLATFORM_PATH)/docker-ptf-nephos.mk
+
+NPX_DIAG = npx_diag
+$(NPX_DIAG)_URL = "ftp://sonic:sonic@172.19.77.245/lib/nephos/npx_diag_0.0.0-20171002"
```
Comment (lguohan) [inline platform/nephos/rules.mk]: this is private ip address, we cannot connect to it.

File: slave.mk [hunk 2]
```diff
@@ -379,7 +379,7 @@ $(DOCKER_LOAD_TARGETS) : $(TARGET_PATH)/%.gz-load : .platform docker-start $$(TA
 ###############################################################################
 
 # targets for building installers with base image
-$(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS)) : $(TARGET_PATH)/% : .platform onie-image.conf $$(addprefix $(DEBS_PATH)/,$$($$*_DEPENDS)) $$(addprefix $(DEBS_PATH)/,$$($$*_INSTALLS)) $$(addprefix $(FILES_PATH)/,$$($$*_FILES)) $(addprefix $(DEBS_PATH)/,$(INITRAMFS_TOOLS) $(LINUX_KERNEL) $(IGB_DRIVER) $(SONIC_DEVICE_DATA) $(SONIC_UTILS)) $$(addprefix $(TARGET_PATH)/,$$($$*_DOCKERS)) $$(addprefix $(PYTHON_WHEELS_PATH)/,$(SONIC_CONFIG_ENGINE))
+$(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS)) : $(TARGET_PATH)/% : .platform onie-image.conf $$(addprefix $(DEBS_PATH)/,$$($$*_DEPENDS)) $$(addprefix $(DEBS_PATH)/,$$($$*_INSTALLS)) $$(addprefix $(FILES_PATH)/,$$($$*_FILES)) $(addprefix $(DEBS_PATH)/,$(INITRAMFS_TOOLS) $(LINUX_KERNEL) $(IGB_DRIVER) $(IXGBE_DRIVER)  $(SONIC_DEVICE_DATA) $(SONIC_UTILS)) $$(addprefix $(TARGET_PATH)/,$$($$*_DOCKERS)) $$(addprefix $(PYTHON_WHEELS_PATH)/,$(SONIC_CONFIG_ENGINE))
```
Comment (lguohan) [inline slave.mk]: extra whitespace.

File: device/ingrasys/x86_64-ingrasys_s9130_32x-r0/INGRASYS-S9130-32X/port_config.ini [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# name          lanes             alias             port
```
Comment (jleveque) [inline device/ingrasys/x86_64-ingrasys_s9130_32x-r0/INGRASYS-S9130-32X/port_config.ini]: Change final column title from `port` to `index`.

Refer to https://github.com/Azure/SONiC/wiki/Porting-Guide.

File: platform/nephos/one-aboot.mk [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# sonic broadcom one image installer
```
Comment (stcheng) [inline platform/nephos/one-aboot.mk]: if only .bin is created, this file is not needed.

File: platform/nephos/one-image.mk [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# sonic broadcom one image installer
```
Comment (stcheng) [inline platform/nephos/one-image.mk]: change the platform name

File: platform/nephos/rules.mk [hunk 6]
```diff
@@ -0,0 +1,30 @@
+include $(PLATFORM_PATH)/sdk.mk
+include $(PLATFORM_PATH)/sai.mk
+include $(PLATFORM_PATH)/platform-modules-ingrasys.mk
+include $(PLATFORM_PATH)/docker-orchagent-nephos.mk
+include $(PLATFORM_PATH)/docker-syncd-nephos.mk
+include $(PLATFORM_PATH)/docker-syncd-nephos-rpc.mk
+include $(PLATFORM_PATH)/one-image.mk
+include $(PLATFORM_PATH)/one-aboot.mk
```
Comment (stcheng) [inline platform/nephos/rules.mk]: aboot is not needed.

File: platform/nephos/rules.mk [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+include $(PLATFORM_PATH)/libsaithrift-dev.mk
+include $(PLATFORM_PATH)/python-saithrift.mk
+include $(PLATFORM_PATH)/docker-ptf-nephos.mk
+
+NPX_DIAG = npx_diag
+$(NPX_DIAG)_URL = "https://github.com/NephosInc/SONiC/raw/master/sdk/npx_diag_2.0.1-20171020"
+
+DSSERVE = dsserve
```
Comment (stcheng) [inline platform/nephos/rules.mk]: is this needed?

File: platform/nephos/one-aboot.mk (hunk 4 above)
Comment (kaiyu22) [reply platform/nephos/one-aboot.mk]: thanks, I will update it.

File: platform/nephos/one-image.mk (hunk 5 above)
Comment (kaiyu22) [reply platform/nephos/one-image.mk]: thanks, I will update it.

File: platform/nephos/rules.mk (hunk 6 above)
Comment (kaiyu22) [reply platform/nephos/rules.mk]: thanks, I will update it.

File: platform/nephos/rules.mk (hunk 7 above)
Comment (kaiyu22) [reply platform/nephos/rules.mk]: Yes, this is needed.
Nephos support the npx_diag utility like bcmcmd to show/debug the ASIC information.

Comment (stcheng) [reply platform/nephos/rules.mk]: thanks

**Pull request merged**